central/sensor/service/service_impl_test.go

```diff
@@ -55,3 +75,216 @@ func TestGetCertExpiryStatus(t *testing.T) {
 		})
 	}
 }
+
+// CRS Test Suite.
+
+type crsTestSuite struct {
+	suite.Suite
+	mockCtrl *gomock.Controller
+	context  context.Context
+}
+
+func (s *crsTestSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+	s.context = sac.WithAllAccess(context.Background())
+	utils.Should(testutils.LoadTestMTLSCerts(s.T()))
+}
+
+func TestCrs(t *testing.T) {
+	suite.Run(t, new(crsTestSuite))
+}
+
+func (s *crsTestSuite) TestCrsCentralReturnsAllServiceCertificates() {
+	sensorService, mockServer := newSensorService(s.context, s.mockCtrl)
+
+	// First-time CRS cluster registration.
+	mockServer.prepareNewHandshake(sensorHello)
+	err := sensorService.Communicate(mockServer)
+	s.NoError(err)
+	s.Len(mockServer.clustersRegistered, 1, "expected exactly one registered cluster")
+	centralHello := retrieveCentralHello(s, mockServer)
+	certBundle := centralHello.GetCertBundle()
+	s.Len(certBundle, 15, "expected 15 entries (1 CA cert, 7 service certs, 7 service keys) in bundle")
```

Comment:
Nit: perhaps extract these two lines into a reusable assertion method to avoid repetition in `TestCrsFlowCanBeRepeated`?

Reply:
done in 5cde41396c90ad3ef0e28d0040b98553be305e27.

Reply:
Maybe `assertCertificateBundleComplete` should be a method on *crsTestSuite, but up to you.

Reply:
No strong feelings on my side, but I thought it's less defining behavior for the `crsTestSuite` type, instead more of a utility function.